### PR TITLE
Marked CookieTempDataProvider's cookie as non-essential

### PR DESCRIPTION
**Pull request by kichalla**
Fixes https://github.com/aspnet/Home/issues/2408 : Determine if the antiforgery cookie is considered essential

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.ViewFeatures/CookieTempDataProviderOptions.cs [hunk 1]
```diff
@@ -18,6 +18,11 @@ public class CookieTempDataProviderOptions
             HttpOnly = true,
             SameSite = SameSiteMode.Strict,
 
+            // This cookie has been markted as non-essential because a user could use the SessionStateTempDataProvider,
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/CookieTempDataProviderOptions.cs]: marked.

File: src/Microsoft.AspNetCore.Mvc.ViewFeatures/CookieTempDataProviderOptions.cs [hunk 2]
```diff
@@ -33,7 +38,12 @@ public class CookieTempDataProviderOptions
         /// <para>
         /// <see cref="CookieBuilder.SameSite"/> defaults to <see cref="SameSiteMode.Strict"/>.
         /// <see cref="CookieBuilder.SecurePolicy"/> defaults to <see cref="CookieSecurePolicy.SameAsRequest" />.
-        /// <see cref="CookieBuilder.HttpOnly"/> defaults to <c>true</c>
+        /// <see cref="CookieBuilder.HttpOnly"/> defaults to <c>true</c>.
+        /// <see cref="CookieBuilder.IsEssential"/> defaults to <c>false</c>, This property is only considered when a
+        /// user opts into the CookiePolicyMiddleware. This middelware, by default, requires consent for all
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/CookieTempDataProviderOptions.cs]: The middleware does not require consent by default, the template will.

File: src/Microsoft.AspNetCore.Mvc.ViewFeatures/CookieTempDataProviderOptions.cs [hunk 3]
```diff
@@ -33,7 +38,12 @@ public class CookieTempDataProviderOptions
...
+        /// non-essential cookies. If you are using this middleware and want to use
+        /// <see cref="CookieTempDataProvider"/>, then either set this property to <c>true</c> or
+        /// provide consent to this cookie.
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/CookieTempDataProviderOptions.cs]: or request user consent for non-essential cookies.

**Comment by kichalla** [comment]
ping

**Pull request merged**